hub/core/tensor.py

```diff
@@ -66,13 +80,18 @@ def _get_metas_from_kwargs(
     return tensor_meta, index_meta
 
 
-def append_tensor(array: np.ndarray, key: str, storage: StorageProvider, **kwargs):
+def append_tensor(
+    sample: Optional[Union[np.ndarray, Dict, float, int]],
+    key: str,
+    storage: StorageProvider,
+    **kwargs,
+):
     """Append to an existing tensor with an array. This array will be chunked and sent to `storage`.
 
     For more on chunking, see the `generate_chunks` method.
 
     Args:
-        array (np.ndarray): Array to be chunked/written. This array will be considered as 1 sample.
+        sample (Union[np.ndarray, Dict, float, int]): Array to be chunked/written. This array will be considered as 1 sample.
```

Comment:
in what case would this be a dict?

Reply:
should have updated to `Sample`, good catch